pkg/keda/triggerauthentication_test.go

```diff
@@ -0,0 +1,385 @@
+			runtimeObjects = append(runtimeObjects, testTriggerAuth)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		builderResult, err := PullTriggerAuthentication(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError != nil {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		} else {
+			assert.Equal(t, testTriggerAuth.Name, builderResult.Object.Name)
+			assert.Equal(t, testTriggerAuth.Namespace, builderResult.Object.Namespace)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestNewTriggerAuthenticationBuilder(t *testing.T) {
+	testCases := []struct {
+		name          string
+		namespace     string
+		expectedError string
+	}{
+		{
+			name:          defaultTriggerAuthName,
+			namespace:     defaultTriggerAuthNamespace,
+			expectedError: "",
+		},
+		{
+			name:          "",
+			namespace:     defaultTriggerAuthNamespace,
+			expectedError: "triggerAuthentication 'name' cannot be empty",
+		},
+		{
+			name:          defaultTriggerAuthName,
+			namespace:     "",
+			expectedError: "triggerAuthentication 'nsname' cannot be empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testSettings := clients.GetTestClients(clients.TestClientParams{})
+		testTriggerAuthBuilder := NewTriggerAuthenticationBuilder(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, testTriggerAuthBuilder.errorMsg)
+		assert.NotNil(t, testTriggerAuthBuilder.Definition)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.name, testTriggerAuthBuilder.Definition.Name)
+			assert.Equal(t, testCase.namespace, testTriggerAuthBuilder.Definition.Namespace)
+		}
+	}
+}
+
+func TestTriggerAuthenticationExists(t *testing.T) {
+	testCases := []struct {
+		testTriggerAuth *TriggerAuthenticationBuilder
+		expectedStatus  bool
+	}{
+		{
+			testTriggerAuth: buildValidTriggerAuthBuilder(buildTriggerAuthClientWithDummyObject()),
+			expectedStatus:  true,
+		},
+		{
+			testTriggerAuth: buildInValidTriggerAuthBuilder(buildTriggerAuthClientWithDummyObject()),
+			expectedStatus:  false,
+		},
+		{
+			testTriggerAuth: buildValidTriggerAuthBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedStatus:  false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.testTriggerAuth.Exists()
+		assert.Equal(t, testCase.expectedStatus, exist)
+	}
+}
+
+func TestTriggerAuthenticationGet(t *testing.T) {
+	testCases := []struct {
+		testTriggerAuth *TriggerAuthenticationBuilder
+		expectedError   error
+	}{
+		{
+			testTriggerAuth: buildValidTriggerAuthBuilder(buildTriggerAuthClientWithDummyObject()),
+			expectedError:   nil,
+		},
+		{
+			testTriggerAuth: buildInValidTriggerAuthBuilder(buildTriggerAuthClientWithDummyObject()),
+			expectedError:   fmt.Errorf("triggerauthentications.keda.sh \"\" not found"),
```

Comment:
It has to failed on different error.
Something like:
 \"\" is invalid: metadata.name: Required value: name is required

Reply:
it does not catch the buildInValidTriggerAuthBuilder execution error since it is not part of the test buddy. We are getting an error of the .Get execution

Reply:
the only option to get some detailed error about empty name for example is to add builder.Definition.Name and builder.Definition.Namespace verification to the Get method. But I didn't fined any Get method in our repo doing so

Reply:
The root cause is missing condition in validate method:

```
	if builder.errorMsg != "" {
		glog.V(100).Infof("The %s builder has error message: %s", resourceCRD, builder.errorMsg)

		return false, fmt.Errorf(builder.errorMsg)
	}
```